tests/system/common.py

```diff
@@ -771,6 +775,35 @@ def running_task_status(task_statuses):
     assert False, "Did not find a TASK_RUNNING status in task statuses: %s" % (task_statuses,)
 
 
+def unreachable_tasks(framework_name='marathon'):
+    """ Returns the unreachable tasks for a given framework.
+    """
+    client = mesos.DCOSClient()
+    frameworks = client.get_master_state()['frameworks']
+    for framework in frameworks:
+        if framework["name"] == framework_name:
+            return framework['unreachable_tasks']
+    return []
+
+
+def unreachable_task_predicate(service):
+    return len(unreachable_tasks(service)) > 0
+
+
+def wait_for_unreachable_task(service='marathon', timeout_sec=10*60, inverse=False):
```

Comment:
I'd much rather work with retry and assert.

Reply:
uses shakedown as a standard with the added benefit of providing stdout feedback for long wait periods.   I prefer to stick with this.

Reply:
Hm, I actually find the shakedown logs rather "polluting". retry with assert gives a little cleaner feedback in the test reports. See https://github.com/dcos/dcos/pull/2426 vs https://github.com/dcos/dcos/pull/2421.

It's not a big deal but I preferr errors such as `assert 'DEGRADED' == 'STABLE' failed` to `TimeoutError`.